src/main/java/org/quizgen/utils/encryption/HashPassword.java

```diff
@@ -1,40 +1,84 @@
 package org.quizgen.utils.encryption;
 
-import org.apache.commons.codec.binary.Hex;
-
-import javax.crypto.SecretKey;
 import javax.crypto.SecretKeyFactory;
 import javax.crypto.spec.PBEKeySpec;
-import java.io.UnsupportedEncodingException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
 import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
 import java.security.spec.InvalidKeySpecException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Optional;
 
-public class HashPassword {
-    public static void main(String[] args) throws UnsupportedEncodingException {
+public class HashPassword{
+    private static final int ITERATIONS = 65536;
+    private static final int KEY_LENGTH = 512;
+    private static final String ALGORITHM = "PBKDF2WithHmacSHA512";
+    private static final SecureRandom RAND = new SecureRandom();
+    public HashPassword() {
+        try {
+            tester();
+        } catch (InvocationTargetException | NoSuchMethodException | InstantiationException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+    public static void main(String[] args) {
+        HashPassword hashPassword = new HashPassword();
+        System.out.println(hashPassword);
```

Comment:
This System.out.println was for testing and printing the hashed password and salt